core/services/fcm/src/main/java/com/mongodb/stitch/core/services/fcm/FcmSendMessageRequest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.stitch.core.services.fcm;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.bson.Document;
+
+/**
+ * An FcmSendMessageRequest encapsulates the details of an FCM send message request.
+ */
+public final class FcmSendMessageRequest {
+  private final FcmSendMessagePriority priority;
+  private final String collapseKey;
+  private final Boolean contentAvailable;
+  private final Boolean mutableContent;
+  private final Long timeToLive;
+  private final Document data;
+  private final FcmSendMessageNotification notification;
+
+  private FcmSendMessageRequest(
+      final FcmSendMessagePriority priority,
+      final String collapseKey,
+      final Boolean contentAvailable,
+      final Boolean mutableContent,
+      final Long timeToLive,
+      final Document data,
+      final FcmSendMessageNotification notification
+  ) {
+    this.priority = priority;
+    this.collapseKey = collapseKey;
+    this.contentAvailable = contentAvailable;
+    this.mutableContent = mutableContent;
+    this.timeToLive = timeToLive;
+    this.data = data;
+    this.notification = notification;
+  }
+
+  /**
+   * Returns the priority of the message.
+   *
+   * @return the priority of the message.
+   */
+  @Nonnull
+  public FcmSendMessagePriority getPriority() {
+    return priority;
+  }
+
+  /**
+   * Returns the group of messages that can be collapsed.
+   *
+   * @return the group of messages that can be collapsed.
+   */
+  @Nullable
+  public String getCollapseKey() {
+    return collapseKey;
+  }
+
+  /**
+   * Returns whether or not to indicate to the client that content is available in order
+   * to wake the device. Note: iOS only.
```

Comment:
[nit] "iOS only" could be a little confusing. Maybe "for messages to iOS devices only". Also applies to other places in this file, and there were also places that said "Android only" that could be a little more clear

Reply:
Ok